File: c++/p6_LargestContinuousArraySum.cpp
/*
Find the largest continuous Array sum of the given Array.
Array will contain negative elements as well.
*/

#include <iostream>
#include <vector>

using namespace std;

int largestContinuousArraySum(vector<int>& nums) {
    int maxSum = nums[0];
    int currentSum = nums[0];

    for (int i = 1; i < nums.size(); i++) {
        currentSum = max(nums[i], currentSum + nums[i]);
        maxSum = max(maxSum, currentSum);
    }

    return maxSum;
}

int main() {
    vector<int> nums = { -2, 1, -3, 4, -1, 2, 1, -5, 4 };
    int maxSum = largestContinuousArraySum(nums);
    
    cout << "Largest Continuous Array Sum: " << maxSum << endl;
    
    return 0;
}

File: python/p6_LargestContinuousArraySum..py
"""
Find the largest continuous Array sum of the given Array.
Array will contain negative elements as well.

!! Do not hard code !!
"""

nums = [ 0, 4, -7, -3, 8, -9, 1, 6, 5, -2 ]  # Output: 12

# Start code from here

def maxSubArray(nums):
    max_sum = float('-inf')
    current_sum = 0
    
    for num in nums:
        current_sum += num

        if max_sum < current_sum:
            max_sum = current_sum
        
        if current_sum < 0:
            current_sum = 0

    return max_sum

result = maxSubArray(nums)
print("Maximum Subarray Sum:",result)
